src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -294,28 +303,32 @@ instance Buildable (TxIn, TxOut) where
     build (txin, txout) = build txin <> " ==> " <> build txout
 
 data TxMeta = TxMeta
-    { txId :: !(Hash "Tx")
-    , depth :: !(Quantity "block" Natural)
+    { txMetaId :: !(Hash "Tx")
     , status :: !TxStatus
     , direction :: !Direction
-    , timestamp :: !Timestamp
     , slotId :: !SlotId
```

Comment:
So, I've removed the `depth` here since it depends on the current chain tip. The timestamp is also a bit tricky because we need to know the epoch length for each previous epoch in order to compute this correctly (timestamp = corresponding UTC value for the beginning of the slot). I've also removed it for now and I suggest to leave the calculation of this one to later.

Reply:
I will still have to have a type to represent these fields for the user to implement return json return type for new transaction but I am assuming this return type will be created in `Wallet.Api.Types` where `Primitive` modules contain minimal data which might be persisted later (and probably will be) :
```
{
  "id": "1423856bc91c49e928f6f30f4e8d665d53eb4ab6028bd0ac971809d514c92db1",
  "amount": {
    "quantity": 42000000,
    "unit": "lovelace"
  },
  "inserted_at": {
    "time": "2019-04-01T19:26:18.474Z",
    "block": {
      "slot_number": 1337,
      "epoch_number": 14
    }
  },
  "depth": {
    "quantity": 0,
    "unit": 1337
  },
  "direction": "outgoing",
  "inputs": [
    {
      "address": "2cWKMJemoBam7gg1y5K2aFDhAm5L8fVc96NfxgcGhdLMFTsToNAU9t5HVdBBQKy4iDswL",
      "amount": {
        "quantity": 42000000,
        "unit": "lovelace"
      }
    }
  ],
  "outputs": [
    {
      "address": "2cWKMJemoBam7gg1y5K2aFDhAm5L8fVc96NfxgcGhdLMFTsToNAU9t5HVdBBQKy4iDswL",
      "amount": {
        "quantity": 42000000,
        "unit": "lovelace"
      }
    }
  ],
  "status": "pending"
}
```

Reply:
Yes indeed, that's where the difference between our API domain and business domain lies.